subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/maven/MavenPomRelocationIntegrationTest.groovy

```diff
@@ -0,0 +1,123 @@
+
+    @Issue('https://github.com/gradle/gradle/issues/1789')
+    @Unroll
+    def "succeed when relocation exists"() {
+        given:
+        def original = publishPomWithRelocation('groupA', 'projectA', relocationGroupId, relocationArtifactId)
+        def newModule = mavenHttpRepo.module(newGroupId, newArtifactId, "1.0").publish()
+
+        and:
+        createBuildFileWithDependency('groupA', 'projectA')
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("${newArtifactId}-1.0.jar")
+
+        where:
+        relocationGroupId | relocationArtifactId | newGroupId  | newArtifactId
+        'groupB'          | 'projectB'           | 'groupB'    | 'projectB'
+        'newGroupA'       | ''                   | 'newGroupA' | 'projectA'
+    }
+
+    def "double-relocation should succeed"() {
+        given:
+        def moduleA = publishPomWithRelocation('groupA', 'projectA', 'groupB', 'projectB')
+        def moduleB = publishPomWithRelocation('groupB', 'projectB', 'groupC', 'projectC')
+        def moduleC = mavenHttpRepo.module('groupC', 'projectC', "1.0").publish()
+
+        and:
+        createBuildFileWithDependency('groupA', 'projectA')
+
+        and:
+        moduleA.pom.expectGet()
+        moduleB.pom.expectGet()
+        moduleC.pom.expectGet()
+        moduleC.artifact.expectGet()
+
+        when:
+        run "retrieve"
+
+        then:
+        file("libs").assertHasDescendants("projectC-1.0.jar")
+    }
+
+    def "fail when artifact in <relocation> doesn't exist"() {
+        given:
+        def original = publishPomWithRelocation('groupA', 'projectA', 'notExist', 'notExist')
+        def newModule = mavenHttpRepo.module("notExist", "notExist", "1.0")
+
+        and:
+        createBuildFileWithDependency('groupA', 'projectA')
+
+        and:
+        original.pom.expectGet()
+        newModule.pom.expectGet()
+        newModule.artifact.expectHead()
+
+        expect:
+        fails "retrieve"
+        failure.assertHasCause('Could not find notExist:notExist:1.0')
+    }
+
+    def createBuildFileWithDependency(String groupId, String artifactId) {
+        buildFile << """
+repositories { maven { url '${mavenHttpRepo.uri}' } }
+configurations { compile }
+dependencies { compile '${groupId}:${artifactId}:1.0' }
+task retrieve(type: Sync) {
+    into 'libs'
+    from configurations.compile
+}
+"""
+    }
+
+    def publishPomWithRelocation(String groupId, String artifactId, String relocationGroupId, String relocationArtifactId) {
+        def module = mavenHttpRepo.module(groupId, artifactId, '1.0').publishPom()
+        def relocation = "<groupId>${relocationGroupId}</groupId>"
+        relocation += relocationArtifactId ? "<artifactId>${relocationArtifactId}</artifactId>" : ''
```

Comment:
I think this comment hasn't be addressed yet. IMHO adding a blank String is logic we don't need. Could you please fix?